test/metrics/metrics_test.go

```diff
@@ -291,6 +294,20 @@ func TestMetricsWhenUsersBanned(t *testing.T) {
 		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait2.ClusterName) // no user on member2
 		metricsAssertion.WaitForMetricDelta(SpacesMetric, 1, "cluster_name", memberAwait.ClusterName)        // space provisioned on member1
 		metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait2.ClusterName)       // no spaces on member2
+
+		// deleting the original usersignup to cleanup the metrics for next tests
+		err = hostAwait.Client.Delete(context.TODO(), userSignup)
+		require.NoError(t, err)
+		// wait for original usersignup to be deleted
+		err = hostAwait.WaitUntilUserSignupDeleted(userSignup.GetName())
+		require.NoError(t, err)
+		err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(userSignup.GetName())
+		require.NoError(t, err)
+		// and verify metrics reset
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait2.ClusterName)
+		metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait.ClusterName)
+		metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait2.ClusterName)
```

Comment:
this shouldn't be needed as all users provisioned are cleaned at the end of the test https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/cleanup/clean.go
But what you could do is to add a wait for cleanup of Spaces in that file as well - something similar as we do for MURs https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/cleanup/clean.go#L144

Reply:
I saw that the provisioned usersignup are cleaned between one test and the other, but still I've noticed that there is a synchronization issue with the metric for space counts being updated by the time next test starts.

ok let me try and create a similar function for spaces and see if it still passes. thanks for your suggestion!